### PR TITLE
feat: core-v3 upgrade

### PR DESCRIPTION
**Pull request by lekhovitsky**
This pull request introduces minor upgrades to adapters to make them compatible with the latest `core-v3` version.

Notable differences in the latter that are accounted for in this pool request are the following:
* `CreditFacade`'s `multicall` now expects adapters' wrapper functions to return two masks (`tokensToEnable` and `tokensToDisable`), `AbstractAdapter`'s `_enable` and `_disable` has been removed
* changes in contracts names and placement within directories

While upgrades to adapters are minor, overall changes to the system are rather drastic, which completely breaks all tests. Thus, they are removed in a separate commit, which should be reverted later when we're ready to fix them.

Besides, the best place to store pool gateways is still being discussed, so they are also removed in yet another to-be-reverted commit until the solution is found.

### CODE REVIEW

File: contracts/adapters/curve/CurveV1_Base.sol [hunk 1]
```diff
@@ -331,63 +387,97 @@ contract CurveV1AdapterBase is AbstractAdapter, ICurveV1Adapter {
...
+    function _add_all_liquidity_one_coin(int128 i, uint256 rateMinRAY)
+        internal
+        returns (uint256 tokensToEnable, uint256 tokensToDisable)
+    {
         address creditAccount = _creditAccount();
 
         address tokenIn = _get_token(i, false);
         uint256 amount = IERC20(tokenIn).balanceOf(creditAccount); // F: [ACV1-9]
-        if (amount <= 1) return;
+        if (amount <= 1) return (0, 0);
```
Comment (0xmikko) [inline contracts/adapters/curve/CurveV1_Base.sol:458]: the same issue wiith disable

File: contracts/adapters/curve/CurveV1_Base.sol [hunk 2]
```diff
@@ -424,71 +514,102 @@ contract CurveV1AdapterBase is AbstractAdapter, ICurveV1Adapter {
     /// @dev Internal implementation of `remove_liquidity`
     ///      - passes calldata to the target contract
     ///      - enables all pool tokens
-    function _remove_liquidity() internal {
+    function _remove_liquidity() internal returns (uint256 tokensToEnable, uint256 tokensToDisable) {
         _execute(msg.data);
-        _changeEnabledTokens(token0Mask | token1Mask | token2Mask | token3Mask, 0); // F: [ACV1_2-5, ACV1_3-5, ACV1_4-5]
+        (tokensToEnable, tokensToDisable) = (token0Mask | token1Mask | token2Mask | token3Mask, 0); // F: [ACV1_2-5, ACV1_3-5, ACV1_4-5]
```
Comment (0xmikko) [inline contracts/adapters/curve/CurveV1_Base.sol:519]: BitMask is also preferale

**Review by 0xmikko — APPROVED** [review]

File: contracts/adapters/curve/CurveV1_Base.sol (hunk 2 above)
Comment (lekhovitsky) [reply contracts/adapters/curve/CurveV1_Base.sol:519]: `BitMask` didn't exist when these changes were made :D

I'll fix in the optimization/cleanup PR, this one is purely to fix compatibility

File: contracts/adapters/compound/CompoundV2_CErc20Adapter.sol [hunk 3]
```diff
@@ -48,21 +48,25 @@ contract CompoundV2_CErc20Adapter is CompoundV2_CTokenAdapter {
...
-        _changeEnabledTokens(cTokenMask, 0);
+        (tokensToEnable, tokensToDisable) = (cTokenMask, 0);
     }
 
     /// @dev Internal implementation of `mintAll`
     ///      - underlying is approved before the call because cToken needs permission to transfer it
     ///      - cToken is enabled after the call
     ///      - underlying is disabled after the call because operation spends the entire balance
-    function _mintAll() internal override returns (uint256 error) {
+    function _mintAll() internal override returns (uint256 tokensToEnable, uint256 tokensToDisable, uint256 error) {
```
[the remark this answers is not in the document]
Comment (lekhovitsky) [reply contracts/adapters/compound/CompoundV2_CErc20Adapter.sol:66]: Compound's stuff

File: contracts/adapters/compound/CompoundV2_CErc20Adapter.sol [hunk 4]
```diff
@@ -72,42 +76,50 @@ contract CompoundV2_CErc20Adapter is CompoundV2_CTokenAdapter {
...
     ///      - cToken is not disabled after the call because operation doesn't spend the entire balance
-    function _redeem(uint256 amount) internal override returns (uint256 error) {
+    function _redeem(uint256 amount)
+        internal
+        override
+        returns (uint256 tokensToEnable, uint256 tokensToDisable, uint256 error)
+    {
         error = abi.decode(_execute(_encodeRedeem(amount)), (uint256));
-        _changeEnabledTokens(tokenMask, 0);
+        (tokensToEnable, tokensToDisable) = (tokenMask, 0);
```
[the remark this answers is not in the document]
Comment (lekhovitsky) [reply contracts/adapters/compound/CompoundV2_CErc20Adapter.sol:92]: true, the problem is that compiler will generate a warning that named return variable `tokensToDisable` is not used, and IMO it's better to keep it named

File: contracts/adapters/aave/AaveV2_WrappedATokenAdapter.sol [hunk 5]
```diff
@@ -136,62 +143,69 @@ contract AaveV2_WrappedATokenAdapter is AbstractAdapter, IAaveV2_WrappedATokenAd
...
-    function _withdrawAll(bool toUnderlying) internal {
+    function _withdrawAll(bool toUnderlying) internal returns (uint256 tokensToEnable, uint256 tokensToDisable) {
         address creditAccount = _creditAccount();
         uint256 balance = IERC20(targetContract).balanceOf(creditAccount);
-        if (balance <= 1) return;
+        if (balance <= 1) return (0, 0);
 
         uint256 shares;
         unchecked {
             shares = balance - 1;
```
[the remark this answers is not in the document]
Comment (lekhovitsky) [reply contracts/adapters/aave/AaveV2_WrappedATokenAdapter.sol:204]: not sure, will remove in the optimization/cleanup PR

File: contracts/adapters/convex/ConvexV1_BaseRewardPool.sol [hunk 6]
```diff
@@ -92,83 +92,115 @@ contract ConvexV1BaseRewardPoolAdapter is AbstractAdapter, IConvexV1BaseRewardPo
...
+        returns (uint256 tokensToEnable, uint256 tokensToDisable)
+    {
+        (tokensToEnable, tokensToDisable) = _stake(msg.data, false); // F: [ACVX1_P-3]
     }
 
     /// @inheritdoc IConvexV1BaseRewardPoolAdapter
-    function stakeAll() external override creditFacadeOnly {
-        _stake(msg.data, true); // F: [ACVX1_P-4]
+    function stakeAll() external override creditFacadeOnly returns (uint256 tokensToEnable, uint256 tokensToDisable) {
+        (tokensToEnable, tokensToDisable) = _stake(msg.data, true); // F: [ACVX1_P-4]
```
[the remark this answers is not in the document]
Comment (lekhovitsky) [reply contracts/adapters/convex/ConvexV1_BaseRewardPool.sol:106]: true, will do in optimization/cleanup PR as well

**Pull request merged**